src/controllers/authentication.js

```diff
@@ -113,7 +113,18 @@ authenticationController.register = async function (req, res) {
 async function addToApprovalQueue(req, userData) {
 	userData.ip = req.ip;
 	await user.addToApprovalQueue(userData);
-	return { message: '[[register:registration-added-to-queue]]' };
+	let message = '[[register:registration-added-to-queue]]';
+	if (meta.config.showAverageApprovalTime) {
+		const times = await db.getListRange('registration:queue:approval:times', 0, -1);
```

Comment:
Instead of storing times in a list which would cause issues as you have thousands of users added to the queue you can just keep an accumulator and calculate the average time from there. For example when accepting a registration you can do

```
const total = await db.incrObjectField('registration:queue:approval:times', 'totalTime', Date.now() - creation_time);
const counter = await db.incrObjectField('registration:queue:approval:times', 'counter', 1);
await db.setObjectField('registration:queue:approval:times', 'average', total / counter);
```

Now all you have to do to show the average is to read the average field from the object instead of loading the entire list and running times.reduce() to calculate.

`const average_time = await db.getObjectField('registration:queue:approval:times', 'average');`

Then calculating

Reply:
I actually forgot I did it this way before pushing it :D

The array reduce method was just for testing (and it helped with some NaN's), but it seems I forgot to write the actual implementation.

Will fix this soon.